### PR TITLE
Shared queues 2

### PR DESCRIPTION
**Pull request by lulf**
@grs This is a refactoring of the address controller to support the 'grouped' address config internally, though the Rest API is still bound to the 'flat' address space. It also puts the addresses to be configured inside a configmap, while still setting the ADDRESS template parameter until the queue scheduler is added.

To handle the grouped address config, the plan is to create a v2.RestService that supports the new grouped schema.

### CODE REVIEW

File: address-controller-lib/src/main/java/enmasse/address/controller/admin/AddressManagerImpl.java [hunk 1]
```diff
@@ -36,46 +38,68 @@
     private final OpenShiftHelper helper;
     private final DestinationClusterGenerator generator;
 
-    public AddressManagerImpl(OpenShiftHelper openshiftHelper, DestinationClusterGenerator generator) {
+    public AddressManagerImpl(OpenShiftHelper openshiftHelper, DestinationClusterGenerator generator, FlavorRepository flavorRepository) {
         this.helper = openshiftHelper;
```
Comment (grs) [inline address-controller-lib/src/main/java/enmasse/address/controller/admin/AddressManagerImpl.java]: Does this new parameter get used?

File: address-controller-lib/src/main/java/enmasse/address/controller/admin/OpenShiftHelper.java [hunk 2]
```diff
@@ -1,38 +1,41 @@
 package enmasse.address.controller.admin;
 
-import enmasse.address.controller.model.AddressType;
+import com.fasterxml.jackson.databind.ObjectMapper;
```
Comment (grs) [inline address-controller-lib/src/main/java/enmasse/address/controller/admin/OpenShiftHelper.java]: Is this a new dependency, or is it already required through resteasy, vertx or other?

File: address-controller-lib/src/main/java/enmasse/address/controller/admin/OpenShiftHelper.java [hunk 3]
```diff
@@ -43,23 +46,61 @@ public OpenShiftClient getClient() {
...
-                String type = labels.get(LabelKeys.ADDRESS_TYPE);
-                String flavor = labels.get(LabelKeys.FLAVOR);
-                Destination destination = new Destination(address, !flavor.isEmpty(), AddressType.TOPIC.name().equals(type), flavor);
-                if (!resourceMap.containsKey(destination)) {
-                    resourceMap.put(destination, new ArrayList<>());
+
+            if (labels != null && labels.containsKey(LabelKeys.GROUP_ID)) {
+                String groupId = labels.get(LabelKeys.GROUP_ID);
+
+                // First time we encounter this group, fetch the address config for it
```
Comment (grs) [inline address-controller-lib/src/main/java/enmasse/address/controller/admin/OpenShiftHelper.java]: How are updates (to the related configmap) after that handled?

File: address-controller-lib/src/main/java/enmasse/address/controller/admin/OpenShiftHelper.java (hunk 2 above)
Comment (lulf) [inline address-controller-lib/src/main/java/enmasse/address/controller/admin/OpenShiftHelper.java]: @grs The jackson dependency has been there for a long time, before we started using resteasy. Resteasy also uses it.

File: address-controller-lib/src/main/java/enmasse/address/controller/admin/OpenShiftHelper.java (hunk 3 above)
Comment (lulf) [inline address-controller-lib/src/main/java/enmasse/address/controller/admin/OpenShiftHelper.java]: This piece of code is run whenever an address config is deployed, so updates to the configmap is retrieved every time. This comment refers to the fact that we have multiple objects (deploymentconfig, persistentvolume claim ++ any that a flavor creates) with a given group id, and we may fetch the config map pointer from any of them when iterating.

File: address-controller-lib/src/main/java/enmasse/address/controller/admin/AddressManagerImpl.java (hunk 1 above)
Comment (lulf) [inline address-controller-lib/src/main/java/enmasse/address/controller/admin/AddressManagerImpl.java]: Pushed a fix, thanks!

File: address-controller-lib/src/main/java/enmasse/address/controller/admin/OpenShiftHelper.java (hunk 2 above)
Comment (grs) [inline address-controller-lib/src/main/java/enmasse/address/controller/admin/OpenShiftHelper.java]: Cool, just checking!

File: address-controller-lib/src/main/java/enmasse/address/controller/admin/OpenShiftHelper.java (hunk 3 above)
Comment (grs) [inline address-controller-lib/src/main/java/enmasse/address/controller/admin/OpenShiftHelper.java]: Thanks for the clarification!

**Pull request merged**